### PR TITLE
rv-ified reorderSamples

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 1]
```diff
@@ -1811,52 +1811,58 @@ class MatrixTable(val hc: HailContext, val metadata: VSMMetadata,
...
 
+      it.map { rv =>
+        rvb.set(rv.region)
+        rvb.start(localRowType)
+        rvb.startStruct()
+        rvb.addField(localRowType, rv, 0) // pk
+        rvb.addField(localRowType, rv, 1) // v
+        rvb.addField(localRowType, rv, 2) // va
+        rvb.startArray(localNSamples) // gs
+        val ur = new UnsafeRow(localRowType, rv)
```
Comment (tpoterba) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: can't you move this outside the `map` and just set it to the region?

Comment (catoverdrive) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: How would you know the offset?

Comment (cseed) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: I think he means the allocation of ur, and then reuse it in the map.  Yes, I can do that.

Comment (tpoterba) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: it looks like UnsafeRow has a `set` method for this case. Wouldn't it work here? Why not if it won't?

Comment (catoverdrive) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: Oh, got it

**Comment by cseed** [comment]
Addressed @tpoterba comment.

**Pull request merged**